packages/veto-app/backend/src/server/index.ts

```diff
@@ -2,18 +2,24 @@ import serveFrontend from './serveFrontend'
 import setupMiddlewares from './middlewares'
 import { Application } from 'express'
 import { Config } from '../utils/config'
+import db from '../db'
 
 export default async function serverSetup(
   app: Application,
   config: Config,
   setupRoutes: (app: Application) => Application,
 ): Promise<void> {
   console.log('Server is starting...')
-  setupMiddlewares(app, config)
-    .then(setupRoutes)
-    // Note: serveFrontend need to be last to not override /api
-    .then(serveFrontend(config))
-    .then((app: Application) => {
-      app.listen(config.port, () => console.log(`${config.appName} is now listening on port ${config.port}!`))
-    })
+  return (
+    setupMiddlewares(app, config)
+      .then(setupRoutes)
+      // Note: serveFrontend need to be last to not override /api
+      .then(serveFrontend(config))
+      .then((app: Application) => {
```

Comment:
Not a blocker but you could follow the chaining pattern to decouple `app` initialization and `db` 

```typescript
.then((app: Application) => (
  app.listen(config.port, () =>
    console.log(`${config.appName} is now listening on port ${config.port}!`),
  )
). then(db.init);
```

If going that route, you would need to return `server` to enable the chain to continue
``` 
(server: Server): ExpressServer => {
  instance = Gun({
    web: server,
    file: config.db.file,
  })

  return server;
}
```

Reply:
Done 👍